### PR TITLE
Set up RSpec test suite and unit testing

### PR DESCRIPTION
**Pull request by Fosome**
This PR is meant to get unit testing around the client code.  

1) Replace old test suite with RSpec3
2) Unit test for `WePay::Client`
3) 100% test coverage with Simplecov
4) Update README.md to reflect new test suite

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,14 +3,12 @@ source "https://rubygems.org"
 gem 'rake'
 
 group :test do
-  gem 'rubyunit'
-  gem "codeclimate-test-reporter", require: false
-  gem 'coveralls', require: false
-  gem 'scrutinizer-ocular', require: false
+  gem 'rspec'
```
Comment (skyzyx) [inline Gemfile]: We're going to want to keep these three lines, as we want to leverage these services just as soon as we get everything using Travis-CI.

```ruby
gem "codeclimate-test-reporter", require: false
gem 'coveralls', require: false
gem 'scrutinizer-ocular', require: false
```

File: spec/we_pay/client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,446 @@
+require 'spec_helper'
```
Comment (skyzyx) [inline spec/we_pay/client_spec.rb]:
```
s/we_pay/wepay/
```

**Comment by skyzyx** [comment]
@Fosome: Once we merge this, we'll need to massage it a little. But overall it looks pretty solid. Let me run through the test suite.

**Comment by skyzyx** [comment]
I'm going to merge it, then need to make some tweaks. But thank you for this PR. :)

**Comment by Fosome** [comment]
@skyzyx Thanks!  I'd love to help however.  I have some more refactoring in mind!

**Pull request merged**